stl/inc/xloctime

```diff
@@ -731,8 +731,12 @@ protected:
 
         for (_Num = 16;; _Num *= 2) { // convert into ever larger string buffer until success
             _Str.append(_Num, '\0');
-            if (0 < (_Count = _Strftime(&_Str[0], _Str.size(), _Fmt, _Pt, _Tnames._Getptr()))) {
+            _Count = _Strftime(&_Str[0], _Str.size(), _Fmt, _Pt, _Tnames._Getptr());
+            if (0 < _Count) {
                 break;
+            } else if (errno == EINVAL) {

```

Comment:
As this occurs in a loop, I believe that we should extract `errno` as we do elsewhere:

https://github.com/microsoft/STL/blob/f675d68f03cfb7a303cd5408502f2642947d32b7/stl/inc/string#L90

What's happening is that `errno` is a macro that involves Thread-Local Storage, so it can have a measurable cost to say over and over. Binding a reference pays the cost once per function call and doesn't add too much verbosity.

Occurs three times in this file.